common/src/main/java/org/apache/uniffle/common/netty/protocol/SendShuffleDataRequest.java

```diff
@@ -103,7 +103,20 @@ private static Map<Integer, List<ShuffleBlockInfo>> decodePartitionData(ByteBuf
       int lengthOfShuffleBlocks = byteBuf.readInt();
       List<ShuffleBlockInfo> shuffleBlockInfoList = Lists.newArrayList();
       for (int j = 0; j < lengthOfShuffleBlocks; j++) {
-        shuffleBlockInfoList.add(Decoders.decodeShuffleBlockInfo(byteBuf));
+        try {
+          shuffleBlockInfoList.add(Decoders.decodeShuffleBlockInfo(byteBuf));
+        } catch (Throwable t) {
+          if (!shuffleBlockInfoList.isEmpty()) {
+            shuffleBlockInfoList.forEach(sbi -> sbi.getData().release());
+          }
```

Comment:
Do we need check empty?

Reply:
The check condition is redundant. I will remove the check.